.devcontainer/docker-compose.yml

```diff
@@ -2,8 +2,8 @@ version: '3'
 services:
   dev:
     build:
-      context: ../
-      dockerfile: .devcontainer/Dockerfile
+      context: ../.devcontainer
```

Comment:
Correct me if I'm wrong, but I don't think we can do this, since we need access to the `flake.nix` in the root for the nix dev shell to get set up

Reply:
Yes, you are correct. I'll rearrange the paths for this.

On the other hand I left a binary reference in the container.json too.
Thanks for the feedback.

Reply:
Hey @rvolosatovs I know why, since the dockerignore did not allow to copy the `nix.conf`.

Either we don't ignore the `.devcontainer` folder in the `.dockerignore` and we can copy the `nix.conf`, or we do it somehow manually instead of copying? - Which I think we shouldn't

The issue is that the COPY [here](https://github.com/wasmCloud/wasmCloud/blob/8872acb22a2e0a9b923876e37117cf22973715be/.devcontainer/Dockerfile#L3) cannot access this file at this time - which is the docker build it only exists in the running container after it started and mounted by the docker compose [here](https://github.com/wasmCloud/wasmCloud/blob/8872acb22a2e0a9b923876e37117cf22973715be/.devcontainer/docker-compose.yml#L11-L12).

Can I remove the `.devcontainer` directory on the top level? Is it harmful in any way other than copying it for dev containers?

Reply:
Not ignoring `.devcontainer` in the root sounds good to me, the root Dockerfile is only really used for development anyway